### DISCUSSION TITLE
pyproject.toml API version specification

### DISCUSSION THREAD

**Discussion by jorenham** [python-poetry/poetry#6569] [Ideas]
At the moment, new poetry releases with backwards-incompatible changes could become very problematic: 

Most users install poetry globally on their system, which would require every single one of their poetry project `pyproject.toml`'s definitions to be compatible with their current poetry version. 
I don't know whether I'm an outlier here, but I have 28 poetry projects on my laptop at the moment 😅 .

In `docker-compose.yml`, you can specify the version of the compose spec: https://docs.docker.com/compose/compose-file/#version-top-level-element. 
Applying that idea for `pyproject.toml`, would allow all projects on a system with a single poetry installation to have different  "poetry spec" versions. 

Any thoughts?

**Comment by neersighted** [comment]
We have discussed adding a `minimum-poetry-version` to pyproject.toml. Currently the only backwards-incompatible change is the introduction of groups, but it is not breaking as you can currently opt out out them. We plan to introduce a `minimum-poetry-version` around the same time that we fully remove the old `dev-` syntax -- a PR would be welcome if this is something you want to work on.

**Reply by jorenham** [reply]
Ah good to hear. 
I'm afraid I don't have the time available at the moment to submit a PR myself at the moment.
